sdk/batch/Microsoft.Azure.Batch/Tools/ConfigureAwaitAnalyzer/AssemblyAttributes.cs

```diff
@@ -3,4 +3,4 @@
 
 ﻿using System.Runtime.CompilerServices;
 
-[assembly: InternalsVisibleTo("ConfigureAwaitAnalyzer.Tests")]
\ No newline at end of file
+[assembly: InternalsVisibleTo("ConfigureAwaitAnalyzer.Tests, PublicKey=0024000004800000940000000602000000240000525341310004000001000100b5fc90e7027f67871e773a8fde8938c81dd402ba65b9201d60593e96c492651e889cc13f1415ebb53fac1131ae0bd333c5ee6021672d9718ea31a8aebd0da0072f25d87dba6fc90ffd598ed4da35e44c398c454307e8e33b8426143daec9f596836f97c8f74750e5975c64e2189f45def46b2a2b1247adc3652bf5c308055da9")]
```

Comment:
Do you absolutely need to InternalsVisibleTo? In general I would recommend against it.

Reply:
We definitely don't absolutely need it (at least not on this project). We use it on quite a few projects though for unit testing purposes...

Even then we probably could change things to where we don't need it, but it's a non-trivial amount of work to do that. I'll make a note of it though and see how much we're actually using it across our projects out of band from this PR.

Reply:
Thanks. If you are interested in some of the reasons I don't recommend it see https://github.com/dotnet/corefx/issues/1604

Reply:
Yeah I just looked and for our main unit test project, the main thing we're using `InternalsVisibleTo` for is to test the interface between our higher-level data plane and the autogenerated code from AutoRest.

For example, we have some tests which generate objects from classes in the AutoRest generated code and ensure that the properties propagate to the higher level interfaces. We also have the same tests in reverse (generate higher level interface types and make sure that things propagate back down to AutoRest).

Since most of the mechanics of this translation is currently internal and we don't have a great way for customers to control/configure it (via interfaces/etc) it's a difficult set of tests to replicate without `InternalsVisibleTo`. In general I agree with the philosophy that we should be testing the public interface, I think in order to do that there though we will need to expand the public interface significantly. It's possible that this would have benefits for customers doing unit testing, etc as well, but obviously it's a larger action item that needs thoughtful design, etc...

For now I'm going to leave it as is, and I've made a note to investigate further in the future.